wrappers/csharp/Managed.cs

```diff
@@ -871,6 +873,44 @@ public static string DecryptWithPasswordAsString(string b64data, string password
             return DecryptWithPasswordAsUtf8String(b64data, password, iterations);
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <returns></returns>
+        public static long JoinShares(byte[][] shares)
+        {
+            var nbShares = shares.Length;
+            var sharesLength = shares[0].Length;
+            var secretLength = (int)Native.JoinSharesSizeNative((UIntPtr)sharesLength);
+            var secret = new byte[secretLength];
+            var tempShares = shares;
+            var sharesTemp = new byte[nbShares][];
+            Array.Copy(shares, sharesTemp, shares.Length);
+            var pointers = Utils.GetArrayReferences(ref shares, nbShares, secretLength);
+            Array.Copy(sharesTemp, shares, sharesTemp.Length);
+            return Native.JoinSharesNative((UIntPtr)nbShares, (UIntPtr)sharesLength, pointers, secret, (UIntPtr)secretLength);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <returns></returns>
+        public static long GenerateSharedKey(int nbShares, int threshold, int secretLength, ref byte[][] shares)
+        {
+            var pointers = Utils.InitializeArray(ref shares, nbShares, secretLength);
+            var result = Native.GenerateSharedKeyNative((UIntPtr)nbShares, (UIntPtr)threshold, (UIntPtr)secretLength, pointers);
+            return result;
+        }
```

Comment:
1. You should free the pointers before returning.
2. The method should return a byte[][], not take it as reference.
3. Native errors are not handled properly, see above comment.